.github/workflows/workflow.yml

```diff
@@ -79,6 +79,45 @@ jobs:
       - run: |
           opam exec -- dune build --profile=dev --only-packages http,cohttp,cohttp-lwt,cohttp-lwt-unix,cohttp-server-lwt-unix,cohttp-mirage,cohttp-async,cohttp-curl-async,cohttp-curl-lwt,cohttp-curl,cohttp-top @install
           opam exec -- dune build -j1 bench/ examples/ cohttp-curl-async/bin/ cohttp-curl-lwt/bin/
-
       - run: opam exec -- dune runtest --profile=dev cohttp cohttp-async cohttp-curl cohttp-curl-async cohttp-curl-lwt cohttp-lwt cohttp-lwt-unix cohttp-mirage cohttp-server-lwt-unix cohttp-top http
 
+  build-test-cohttp-eio:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+        ocaml-compiler:
+          - ocaml-variants.5.0.0+trunk
+        local-packages:
+          - |
+            http.opam
+            cohttp-eio.opam
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - run: sudo apt-get install ncat
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+
+      - run: brew install nmap
+        if: ${{ matrix.os == 'macos-latest' }}
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+          dune-cache: ${{ matrix.os == 'ubuntu-latest' }}
+          opam-local-packages: ${{ matrix.local-packages }}
+
+      - run: opam repo add alpha git+https://github.com/kit-ty-kate/opam-alpha-repository.git && opam update
+
+      - run: opam install cstruct eio_main bigstringaf fmt mdx
```

Comment:
```suggestion
      - run: opam install --with-test --deps-only cohttp-eio
```

Reply:
This won't work with opam files that have  `(http (= :version))` constraint since setup-ocaml uses opam 2.0.